pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2MetadataGenerator.java

```diff
@@ -119,7 +118,7 @@
     public MetadataResolver buildMetadataResolver(final Resource metadataResource) throws Exception {
         final AbstractBatchMetadataResolver resolver;
         if (metadataResource != null) {
-            resolver = new FilesystemMetadataResolver(metadataResource.getFile());
+            resolver = new SpringResourceMetadataResolver(metadataResource);
```

Comment:
What is the use case here? Could you explain the scenario that has you define and load SP metadata that is not backed by a file? What specific implementation of the `WritableResource` are you dealing with?